bot/client/client.ts

```diff
@@ -0,0 +1,57 @@
+import type { RESTPostAPIApplicationCommandsJSONBody } from "../deps.ts";
+import { contentType } from "../deps.ts";
+
+const DISCORD_API_URL = "https://discord.com/api/v8";
+
+interface RegisterInit {
+  botID: string;
+  botToken: string;
+  app: RESTPostAPIApplicationCommandsJSONBody;
+}
+
+/**
+ * overwrite overwrites the Discord Slash Commands associated with this server.
+ *
+ * Based on this cURL command:
+ * ```bash
+ * BOT_TOKEN='replace_me_with_bot_token'
+ * CLIENT_ID='replace_me_with_client_id'
+ * curl -X POST \
+ * -H 'Content-Type: application/json' \
+ * -H "Authorization: Bot $BOT_TOKEN" \
+ * -d '{"name":"hello","description":"Greet a person","options":[{"name":"name","description":"The name of the person","type":3,"required":true}]}' \
+ * "https://discord.com/api/v8/applications/$CLIENT_ID/commands"
+ * ```
+ */
+export async function overwrite(
+  { botID, botToken, app }: RegisterInit,
+): Promise<Response> {
+  const url = makeRegisterGuildCommandsURL(botID);
+  const r = await fetch(url.toString(), {
```

Comment:
Additionally clear the commands that were already registered before adding the latest ones.

Reply:
This can instead be done by the bot developer or admin manually since the application ID is required anyway.